### PR TITLE
Kotlin 2 support

### PR DESCRIPTION
**Pull request by glureau**
K2 doesn't support to have generated code in jvmTest + commonTest. Here is an attempt to **move** those files in commonTest instead of playing with sourceSets.

The move is done just after files are written. Due to KSP incremental building implementation, we need to keep files in jvmTest, so to avoid duplication we're copying the content in commonTest and empty the original jvmTest files.

Versions doesn't match your internal gradle settings plugin, feel free to adjust that draft ;)

Thanks for your time! You rocks guys!

### CODE REVIEW

**Comment by flipper09112** [comment]
This PR is for fix this problem #78 ?

**Comment by glureau** [comment]
Yep, also I can't guarantee it'll always work well: What I'm doing is moving files from jvmTest to commonTest, and I record them in KSP CodeGeneratorImpl, which is not supposed to be used that way. CodeGeneratorImpl is checking explicitely that a file is written in the current generation directory (jvmTest), I bypass that by loading the "internal" KSP lib and directly add in the generated file mutable map. As a result, it's likely to have gradle cache inconsistency, you may have to "./gradlew clean" to remove previously generated files when code is changing.

I'm running that in our KMP app, roughly 150k lines, no problem on CI, but we notice a few edge cases when doing incremental builds with wrong cache invalidation (and a build error when a FakeProvider is setup on commonMain on a class without empty constructor).

TLDR: try and tell me if you see something weird :)

**Comment by SalomonBrys** [comment]
Thanks a lot for the work, but I've decided to pursue a less hacky (and I think more future-proof) way for MocKMP.
Version 2.0.0 has just been released and brings support for Kotlin 2 (with breaking changes, see [Migration guide](https://kosi-libs.org/mockmp/2.0/migration/1to2.html)).
